### DISCUSSION TITLE
up-modal vs up-layer

### DISCUSSION THREAD

**Discussion by bbende** [unpoly/unpoly#297] [Q&A] [answered]
Hello, Just looking at Unpoly 2.x and layers for the first time...

When replacing usage of `up-modal=".my-selector"` to `up-layer="new"`, is there a new way to specify the selector to get the content for the modal?

It seems to be working fine with just `up-layer="new"`, but in this case the response for the href purposely only includes the content for the modal and does not include the full layout, so I assume it is taking the whole body and it just works in this case.

**Comment by adam12** [comment, marked as answer]
Pair `up-layer` with [`up-target`](https://unpoly.com/a-up-follow#up-target).

**Reply by bbende** [reply]
Thanks!

**Comment by triskweline** [comment]
If you're using [`unpoly-migrate.js`](https://unpoly.com/changes/upgrading), your log will show the new usage for all deprecated APIs that you use in your app.
